test/app_profiler/profile_id_test.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+module AppProfiler
+  class ProfileIdTest < TestCase
+    test "ProfileId memoization" do
+      ProfileId.reset
+
+      SecureRandom.expects(:hex).once.returns("profile-id")
+      id = ProfileId.current
+      assert_equal(id, ProfileId.current)
+      assert_equal(id, ProfileId.current)
```

Comment:
```suggestion
      assert_same(id, ProfileId.current)
```
And then remove the first equal assert.